_posts/2020-02-7-dns-based-discovery.md

```diff
@@ -0,0 +1,94 @@
+90.133.4.73
+176.191.182.3
+109.207.166.232
+45.5.117.59
+178.211.170.2
+160.16.0.30
+```
+
+The above displays the result of querying on of these DNS seeds. All the nodes are stored as [`A` records](https://simpledns.plus/help/a-records) for the given domain name. This is quite a simple solution which bitcoin almost soley relies on since removing the [IRC bootstrapping method in v0.8.2](https://en.bitcoin.it/wiki/Network#IRC).
+
+So what makes this DNS based discovery useful? Well, it allows us to have a mutable list of bootstrap nodes without needing to ship a new version of the client every time a list is mutated. It also allows for a more lightweight method of discovering nodes, something very important for resource restricted devices.
+
+Additionally, DNS provides us with a robust and scalable infrastructure. This is due to its hierarchical architecutre. This hierarchical architecture also already makes it distributed such that the failure of one DNS server does not result in us no longer being able to resolve our name.
+
+As with every solution though, there is a trade-off. By storing the list in DNS name an adversary would simply need to censor the DNS records for a specific name. This would prevent any new client trying to join the network from not being able to do so.
+
+One thing you notice when looking at [EIP-1459](https://eips.ethereum.org/EIPS/eip-1459) is that it is a lot more technically complex than bitcoin's way of doing this. So if bitcoin uses this simple method and has proven that it works, why did we need a new method?
+
+There are multiple reasons, but the main one is **security**. In the bitcoin example, an attacker could create a new list and no one querying would be able to tell. This is however mitigated in [EIP-1459](https://eips.ethereum.org/EIPS/eip-1459) where we can verify the integrity of the entire returned list by storing an entire merkle tree in the DNS records.
+
+Let's dive into this. Firstly, a client that is using these DNS records for discovery must know the public key corresponding to the private key controlled by the entity creating the list. This is because the entire list is signed using a secp256k1 private key, giving the client the ability to authenticate the list and know that it has not been tampered with by some external party.
+
+So that already makes this a lot safer than the method bitcoin uses. But how are these lists even stored? As previously stated they are stored using **merkle trees** as follows:
+ - The root of the tree is stored in a [`TXT` record](https://simpledns.plus/help/txt-records), this record contains the tree's root hash, a sequence number which is incremented every time the tree is updated and a signature as stated above. 
+ 
+    Additionally, there is also a root hash to a second tree, a link tree that contains the information to different lists. This link tree allows us to delegate trust and build a graph of multiple merkle trees stored across multiple DNS names.
```

Comment:
Different paragraph formatting?

'link tree' a specific term? I'd link this or italicize/bold, or called it 'linked tree' if we want to describe it